### PR TITLE
Hypernova/crossplatform

### PR DESCRIPTION
**Pull request by chasester**
Basic approval

### CODE REVIEW

**Comment by api-Hypernova** [comment]
Let's not merge this yet until I can get something stable. It still uses some hacks which I don't think are suitable for master yet.

**Comment by chasester** [comment]
is that what that thingy does 0.0 
i was just messing around with shit again :+1:

**Review by api-Hypernova — COMMENTED** [review]
Seems fine, go ahead.

File: source/shared/tools.h [hunk 1]
```diff
@@ -20,11 +20,11 @@ typedef unsigned long long int ullong;
 #define GCC_DOUBLECONCATINATION_FIX1(fun,suffix,under)  GCC_DOUBLECONCATINATION_FIX2(fun,suffix,under)
 #define GCC_DOUBLECONCATINATION_FIX(fun,suffix,under) GCC_DOUBLECONCATINATION_FIX1(fun,suffix,under)
 
-#ifdef _DEBUG
-#define ASSERT(c) assert(c)
-#else
+//#ifdef _DEBUG
+//#define ASSERT(c) assert(c)
+//#else
```
Comment (chasester) [inline source/shared/tools.h:25]: we need to change this back

File: source/shared/wings.cpp [hunk 2]
```diff
@@ -406,18 +406,18 @@ void config(asIScriptEngine *asEngine){
...
 
-	r = asEngine->RegisterObjectProperty("node", "string name", asOFFSET(node, name));
+    //r = asEngine->RegisterObjectProperty("node", "string name", asOFFSET(node, name));
 	r = asEngine->RegisterObjectProperty("node", "vec o", asOFFSET(node, o));
 	r = asEngine->RegisterObjectProperty("node", "vec rot", asOFFSET(node, rot));
 	//r = asEngine->RegisterObjectMethod("node", "void seteditbox(const vec &in)", asMETHOD(node, seteditbox), asCALL_THISCALL); assert(r >= 0);
     //r = asEngine->RegisterObjectMethod("node", "void seteditbox(int i)", asMETHOD(node, seteditbox), asCALL_THISCALL); assert(r >= 0);
 	//rigid body and physics
-    r = asEngine->RegisterObjectType("rigidbody", sizeof(rigidbody), asOBJ_VALUE | asOBJ_APP_CLASS_CDAK); assert(r >= 0);
+    //r = asEngine->RegisterObjectType("rigidbody", sizeof(rigidbody), asOBJ_VALUE | asOBJ_APP_CLASS_CDAK); assert(r >= 0);
```
Comment (chasester) [inline source/shared/wings.cpp]: delete this line we wont need it

File: source/shared/wings.cpp [hunk 3]
```diff
@@ -406,18 +406,18 @@ void config(asIScriptEngine *asEngine){
 	r = asEngine->RegisterObjectBehaviour("node", asBEHAVE_RELEASE, "void f()", asMETHOD(node, release), asCALL_THISCALL); assert(r >= 0);
     //r = asEngine->RegisterObjectBehaviour("node", asBEHAVE_GET_WEAKREF_FLAG, "int &f()", asMETHOD(node, getweakflagref), asCALL_THISCALL); assert(r >= 0);
 
-	r = asEngine->RegisterObjectProperty("node", "string name", asOFFSET(node, name));
+    //r = asEngine->RegisterObjectProperty("node", "string name", asOFFSET(node, name));
```
Comment (chasester) [inline source/shared/wings.cpp]: will need to add back when we fix the string issue or nodes wont init properly

File: source/shared/wings.cpp [hunk 4]
```diff
@@ -406,18 +406,18 @@ void config(asIScriptEngine *asEngine){
...
+    //r = asEngine->RegisterObjectType("rigidbody", sizeof(rigidbody), asOBJ_VALUE | asOBJ_APP_CLASS_CDAK); assert(r >= 0);
     r = asEngine->RegisterObjectBehaviour("rigidbody", asBEHAVE_CONSTRUCT, "void f()", asFUNCTION(BaseConstructor<rigidbody>), asCALL_GENERIC); assert(r >= 0);
     r = asEngine->RegisterObjectBehaviour("rigidbody", asBEHAVE_DESTRUCT, "void f()", asFUNCTION(DeConstructor<rigidbody>), asCALL_GENERIC); assert(r >= 0);
     r = asEngine->RegisterObjectMethod("rigidbody", "void aabbfrommodel(int index)", asMETHOD(rigidbody, rigidbody::aabbfrommodel), asCALL_THISCALL); assert(r >= 0);
     r = asEngine->RegisterGlobalFunction("void addrigidbody(const rigidbody &in, const node @+ to)", asFUNCTIONPR(addrigidbody, (rigidbody&, node*), void), asCALL_CDECL); asFATAL(r);
-	r = asEngine->RegisterGlobalFunction("void createobject(string s, const vec &in, const vec &in)", asFUNCTIONPR(createobject, (str, vec&, vec&), void), asCALL_CDECL); asFATAL(r);
+    //r = asEngine->RegisterGlobalFunction("void createobject(string s, const vec &in, const vec &in)", asFUNCTIONPR(createobject, (str, vec&, vec&), void), asCALL_CDECL); asFATAL(r);
```
Comment (chasester) [inline source/shared/wings.cpp]: will need to add back when we fix the string issue or nodes wont init properly

**Comment by chasester** [comment]
we need to come up with one gitignore list.

File: source/Makefile [hunk 5]
```diff
@@ -71,7 +71,6 @@ CLIENT_OBJS= \
 	shared/scriptbuilder.o\
 	shared/scripthandle.o\
 	shared/scripthelper.o\
-	shared/scriptstring.o\
```
Comment (chasester) [inline source/Makefile]: doesn exist it is just miss spelled, should be scriptstringd.o

**Comment by mufeedali** [comment]
nightglare here, are you done with the clean-ups yet?

**Comment by chasester** [comment]
Depricated commits, we can use this for reference, in the future for additiona cross platform support and changes, but all othese changes would not reflect properly onto the current version of the project.

**Pull request closed without merging**